File: backend/app/schemas/document.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class DocumentCreate(BaseModel):
    title: str
    total_pages: int = 1


class PageItem(BaseModel):
    id: uuid.UUID
    page_number: int
    status: str
    enh_image_path: str | None = None

    model_config = ConfigDict(from_attributes=True)


class DocumentResponse(BaseModel):
    id: uuid.UUID
    title: str
    doc_type: str
    original_filename: str
    total_pages: int
    created_at: datetime
    pages: list[PageItem] = []

    model_config = ConfigDict(from_attributes=True)


class DocumentList(BaseModel):
    id: uuid.UUID
    title: str
    doc_type: str
    original_filename: str
    total_pages: int
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/app/models/document.ts
export interface PageItem {
  id: string;
  page_number: number;
  status: 'pending' | 'processing' | 'completed' | 'failed';
  enh_image_path?: string | null;
}

export interface DocumentResponse {
  id: string;
  title: string;
  doc_type: 'pdf' | 'image';
  original_filename: string;
  total_pages: number;
  created_at: string;
  pages: PageItem[];
}

export interface DocumentList {
  id: string;
  title: string;
  doc_type: 'pdf' | 'image';
  original_filename: string;
  total_pages: number;
  created_at: string;
}

export interface PageResponse {
  id: string;
  document_id: string;
  page_number: number;
  status: string;
  orig_image_path?: string | null;
  enh_image_path?: string | null;
  ocr_text?: string | null;
}

export interface PageStatus {
  id: string;
  status: string;
  enh_image_path?: string | null;
}
